### ISSUE TITLE
"Id" is not an acronym

### ISSUE THREAD

**Issue by jameshfisher** [golang/lint#216]
`golint` gives me warnings like this:

```
foo/bar.go:6:2: struct field FooId should be FooID
```

"Id" is not an acronym and it is intentionally written "Id". It also plays more nicely with CamelCase.

This warning is cluttering up a list of more serious warnings. I would prefer for it to be removed, or for me to be able to silence it.

**Comment by kkirsche** [comment]
According to [Merriam-Webster](http://www.merriam-webster.com/dictionary/ID%5B2%5D), the abbreviation is "ID" which is shortened from Identification.

**Comment by jameshfisher** [comment]
@kkirsche where the top two answers are polar opposites and the thread is full of disagreement. I don't want to get into the argument and I feel like the linter is trying to drag me into it. There are more serious warnings that I want to see, and this silly warning gets in the way of them.

**Comment by peterbourgon** [comment]
That ID is written ID is longstanding Go form. Just accept it and move on to more interesting problems :)

**Comment by jameshfisher** [comment]
@peterbourgon exactly - I want to deal with the interesting real warnings rather than this silly one. I eventually did that by turning off `golang/lint`

**Comment by peterbourgon** [comment]
This isn't a silly warning. You should be writing `ID`. That is accepted Go style.

**Comment by dsnet** [comment]
Part of styling is that it is guided by what the community ends up doing. Using `ID` as the abbreviation for Identity rather than `Id` is what the Go community has been doing and been encouraged to do up until now. There are certainly cases (even in the standard library) where `Id` is used, but those can't be changed due to compatibility agreement.

I believe the worse thing that we can do is to flip-flop on this decision. Regardless of the merits of `ID` or `Id`, we should avoid changing this.
